android/src/main/java/com/wultra/android/powerauth/reactnative/PowerAuthRNModule.java

```diff
@@ -364,26 +364,28 @@ public void commitActivation(String instanceId, final ReadableMap authMap, final
         this.usePowerAuthOnMainThread(instanceId, promise, new PowerAuthBlock() {
             @Override
             public void run(@NonNull PowerAuthSDK sdk) {
-                PowerAuthAuthentication auth = constructAuthentication(authMap);
-                if (auth.usePassword == null) {
-                    promise.reject(EC_PASSWORD_NOT_SET, "Password is not set.");
+                final PowerAuthAuthentication auth = constructAuthentication(authMap, true, promise);
+                if (auth == null) {
```

Comment:
Rather than checking for null (without the promise fulfilling), the `constructAuthentication` could throw an exception that could be caught in the `usePowerAuth` block.

It would also remove a lot of repetitive code - the null check.

Reply:
Good point